File: ai-backend/src/services/user/user_schema.py
from pydantic import BaseModel, EmailStr


class TokenRefreshSchema(BaseModel):
    refresh_token: str


class TokenVerifySchema(BaseModel):
    token: str


class UserLoginSchema(BaseModel):
    email: EmailStr
    password: str


class UserCreateSchema(BaseModel):
    email: EmailStr
    password: str


class UserGoogleSchema(BaseModel):
    email: EmailStr
    name: str
    google_id: str


class UserSchema(BaseModel):
    id: int
    email: EmailStr
    name: str

    class Config:
        from_attributes = True

File: analyst-ui/services/user/user.schema.ts
import { z } from "zod";

export const UserBaseSchema = z.object({
  email: z.string(),
  name: z.string(),
});

export const UserCreateSchema = UserBaseSchema.extend({
  password: z.string(),
});

export const UserSchema = UserBaseSchema.extend({
  id: z.string(),
});

export type User = z.infer<typeof UserSchema>;
